src/types.d.ts

```diff
@@ -99,52 +110,34 @@ export type RuleEvent = {
 	internalLog?: string;
 }
 
+/**
+ * All ESLint-like rules are defined such that ESLint can run them, but the engines have _exactly_ enough difference in
+ * their type definitions that the TS compiler complains if you try to cast one to another. So we use this and {@link ESRule}
+ * as super-types that we can use for our purposes, without worrying about a rule's true engine-specific type.
+ */
 export type ESRuleMetadata = {
 	deprecated?: boolean;
 	// This is optional because ESLint-LWC rules don't have a type parameter set. We'll be defaulting to 'problem' in that case.
 	type?: string;
-	docs: {
-		description: string;
-		recommended: boolean|string;
+	docs?: {
+		description?: string;
+		recommended?: boolean|string;
 		extendsBaseRule?: boolean|string;
-		url: string;
+		url?: string;
 	};
-	/* eslint-disable @typescript-eslint/no-explicit-any */
-	schema: Record<string, any>[];
+	schema?: JSONSchema4|JSONSchema4[];
 }
 
 /**
- * Type mapping to rules returned from eslint
+ * All ESLint-like rules are defined such that ESLint can run them, but the engines have _exactly_ enough difference in
+ * their type definitions that the TS compiler complains if you try to cast one to another. So we use this and {@link ESRuleMetadata}
+ * as super-types that we can use for our purposes, without worrying about a rule's true engine-specific type.
  */
 export type ESRule = {
-	meta: ESRuleMetadata;
+	meta?: ESRuleMetadata;
 	create: (LooseObject) => LooseObject;
 }
 
-/**
- * Type mapping for eslint's results within a report
- */
-export type ESResult = {
-	filePath: string;
-	messages: ESMessage[];
-}
-
-/**
- * Type mapping to report messages output by eslint
- */
-export type ESMessage = {
```

Comment:
These types didn't quite align with the `ESLint.LintResult` and `Linter.LintMessage[]` types, so I just swapped those types in for these, rendering these ones unnecessary.